spec/features/user_edits_information_spec.rb

```diff
@@ -0,0 +1,102 @@
+
+  before do
+    visit new_user_session_path
+    fill_in 'Email', with: user.email
+    fill_in 'Password', with: user.password
+    click_button 'Log in'
+  end
+
+  scenario 'user changes password and email' do
+    click_link 'Account Settings'
+    expect(current_path).to eq edit_user_registration_path
+
+    new_password = "boogeyman"
+    new_email = "boogeyman@gmail.com"
+    fill_in 'Password', with: new_password
+    fill_in 'Password confirmation', with: new_password
+    fill_in 'Email', with: new_email
+    fill_in 'Current password', with: user.password
+    click_button 'Update'
+
+    expect(page).to have_content('Your account has been updated successfully.')
+    expect(page).to have_content(new_email)
+
+    click_link 'Sign Out'
+    visit new_user_session_path
+    fill_in 'Email', with: new_email
+    fill_in 'Password', with: new_password
+    click_button 'Log in'
+    expect(page).to have_content('Signed in successfully')
+  end
+
+  scenario 'password and password confirmation have to match to update' do
+    click_link 'Account Settings'
+    expect(current_path).to eq edit_user_registration_path
+
+    new_password = "boogeyman"
+    fill_in 'Password', with: new_password
+    fill_in 'Password confirmation', with: "boogeymmm"
+    fill_in 'Current password', with: user.password
+    click_button 'Update'
+
+    expect(page).to have_content("Password confirmation doesn't match Password")
+  end
+
+  scenario 'user changes username and avatar' do
+    click_link 'Account Settings'
+    expect(current_path).to eq edit_user_registration_path
+
+    old_username = user.username
+    new_username = 'trishboogeybogey'
+    new_avatar = 'https://scontent.cdninstagram.com/hphotos-xpf1/t51.2885-15/s320x320/e15/928946_916321521769110_83206274_n.jpg'
+    fill_in 'Username', with: new_username
+    fill_in 'Avatar', with: new_avatar
+    fill_in 'Current password', with: user.password
+    click_button 'Update'
+
+    expect(page).to have_content('Your account has been updated successfully.')
+    expect(page).to_not have_content(old_username)
+    expect(page).to have_content(new_username)
+    expect(page).to have_content(new_avatar)
+  end
+
+  scenario 'user cannot access users#edit if not authenticated' do
+    click_link 'Sign Out'
+    visit edit_user_registration_path
+
+    expect(page).to have_content('need to sign in or sign up before continuing')
+  end
+
+  scenario "user cannot edit other users' information" do
+    other_password = "blurryblah"
+    other_user = FactoryGirl.create(:user,
+      password: other_password, password_confirmation: other_password)
+    visit edit_user_registration_path
+    fill_in 'Username', with: 'bubblyblob'
+    fill_in 'Email', with: other_user.email
+    fill_in 'Current password', with: other_user.password
+    click_button 'Update'
+
+    expect(page).to have_content('Email has already been taken')
+    expect(page).to have_content('Current password is invalid')
+  end
```

Comment:
I don't think this is actually testing what you describe in the `scenario` text above. It looks like you're trying to edit the current user's information to match that of the other user, which is a fine thing to test. However, I also want a test that makes sure that the current user cannot visit the `edit` page for another user. This might require the use of some `before_action`s in your controllers - look into those! (See if you can find info on `authorize_admin` or something related)